src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Resolve.java

```diff
@@ -4174,16 +4175,10 @@ JCDiagnostic getDiagnostic(JCDiagnostic.DiagnosticType dkind,
                 return null;
 
             Pair<Symbol, JCDiagnostic> c = errCandidate();
-            if (compactMethodDiags) {
```

Comment:
I believe the issue is here - the old code made the rewriting dependent on the value of `compactMethodDiags`, which in turns looked at which diagnostic formatter was installed. I think we should either move this check in `Log` (when we decide if we need to rewrite), or don't attach a rewriter if the `compactMethodDiags` is not set. I think it would be preferable to keep existing diagnostics as they are (and maybe cleanup the behavior of `compactMethodDiags` later, as an orthogonal fix).